src/analysis.zig

```diff
@@ -3097,6 +3097,11 @@ pub const Type = struct {
         };
     }
 
+    /// If this returns false, the type cannot have the static modifier
+    pub fn isTypeOrFunc(self: Type) bool {
+        return self.is_type_val or self.isFunc() or self.isTypeFunc();
```

Comment:
The `self.isTypeFunc()` is unnecessary and the doc comment should talks about the static modifier even though the function itself has nothing to do with it.

Remove this function and use `self.is_type_val or self.isFunc()` directly.